osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObject.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Catch.Skinning.Default;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Objects.Drawables
+{
+    /// <summary>
+    /// Represents a <see cref="PalpableCatchHitObject"/> caught by the catcher.
+    /// </summary>
+    [Cached(typeof(IHasCatchObjectState))]
+    public abstract class CaughtObject : SkinnableDrawable, IHasCatchObjectState
+    {
+        public PalpableCatchHitObject HitObject { get; private set; }
+        public Bindable<Color4> AccentColour { get; } = new Bindable<Color4>();
+        public Bindable<bool> HyperDash { get; } = new Bindable<bool>();
+
+        /// <summary>
+        /// Whether this hit object should stay on the catcher plate when the object is caught by the catcher.
+        /// </summary>
+        public virtual bool StaysOnPlate => true;
+
+        public override bool RemoveWhenNotAlive => true;
+
+        protected CaughtObject(CatchSkinComponents skinComponent, Func<ISkinComponent, Drawable> defaultImplementation)
+            : base(new CatchSkinComponent(skinComponent), defaultImplementation)
+        {
+            Origin = Anchor.Centre;
+
+            RelativeSizeAxes = Axes.None;
+            Size = new Vector2(CatchHitObject.OBJECT_RADIUS * 2);
+        }
+
+        /// <summary>
+        /// Copies the hit object visual state from another <see cref="IHasCatchObjectState"/> object.
+        /// </summary>
+        public virtual void CopyStateFrom(IHasCatchObjectState objectState)
+        {
+            HitObject = objectState.HitObject;
+            Scale = objectState.Scale;
+            Rotation = objectState.Rotation;
+            AccentColour.Value = objectState.AccentColour.Value;
+            HyperDash.Value = objectState.HyperDash.Value;
+        }
+
+        protected override void FreeAfterUse()
+        {
+            ClearTransforms();
+
+            Alpha = 1;
+            LifetimeStart = double.MinValue;
+            LifetimeEnd = double.MaxValue;
```

Comment:
Lifetime is set by `Expire` when a caught object is dropped. If lifetime is not reset issues will occur.
I removed assignment of `LifetimeStart` because that was not doing anything (I thought the object will be removed when a replay is rewound but it is not, and I added object removing logic to `OnRevertResult`)

Reply:
I somehow thought lifetime had to reset but apparently it is reset by `DrawablePool` and the lines above is not necessary. I think I misunderstood a different bug (that I solved).